sys/exec.c

```diff
@@ -86,13 +87,16 @@ int do_exec(const exec_args_t *args) {
     return -ENOEXEC;
   }
 
-  /* TODO: Get current process description structure */
+  thread_t *td = thread_self();
 
-  /* The current vmap should be taken from the process description! */
-  vm_map_t *old_vmap = get_user_vm_map();
-  /* We may not destroy the current vm map, because exec can still
-   * fail, and in that case we must be able to return to the
-   * original address space
+  /* Please don't switch to other threads while we prepare the address
```

Comment:
I think it does fix the problem, but since #103 entirely messes up my way of imagining the distinction between user and kernel threads, I can't clearly see if it does not have any fancy side-effects.